### DISCUSSION TITLE
feat: (Button) Go to nearest call for help button

### DISCUSSION THREAD

**Discussion by kaganbuzcu** [acikkaynak/deprem-yardim-frontend#547] [Ideas]
## İsteğiniz şeyi tanımlayın 
The task of determining the nearest help call by comparing the user location with the help call data and moving the map center to the relevant location.

** discord name: @kaganbu#7797 **

## Değerlendirdiğiniz alternatifleri tanımlayın
In order to easily direct the teams carrying out rescue activities in the disaster area to the nearest call.

* Sağ üste(mesela), "En Yakınımdaki Yardım Çağrısına Git" görevini uygulayacak bir button eklenmesi 
* Kullanıcı konumunu dönen bir helper yazılması
* Alınan kullanıcı konumu verisinin points ile kıyaslanması ve en yakındaki yardım çağrısı koordinatlarının state'e yazılması
* Tespit edilen en yakın koordinata map center'ın set edilmesi

## Ek Bağlam

Konum verisi talep etmenin yasal yükümlülükleri hakkında bilgim yok, problem yaratabilecek bir konu değilse veya feature'ın kullanışsız olduğu düşünülmüyorsa aşağıdaki gereksinimlere ihtiyaç doğacaktır.

* Wording, buton görünümü ve konumu konularında UI/UX desteği
* Yazılmış kodun revizesi için biraz zaman

![afet-harita-nearest-button](https://user-images.githubusercontent.com/63363182/217695062-728b98a1-d5c7-48e6-804a-d2bc64cc8cd4.gif)

**Comment by 0xCakin** [comment]
@kaganbuzcu Google Maps yol tarifini yol verisi uzerinden saglar. Deprem sirasinda yollar da bozulacagi icin kullanisli bir feature olmaz. Onun yerine en yakin 5 noktayi isaretle olabilir. Bu konuyu discussion'a tasiyalim.

**Reply by kaganbuzcu** [reply]
Geri bildiriminiz için teşekkür ederim. Öncelikle bu, nokta işaretleme veya yol tarifi içeren bir öneri değil, sadece harita merkezini değiştirecek bir öneriydi. Bütün çağrılar harita üzerinde pinli zaten ve pin'e tıklandığında drawer'da Google Maps yol tarifi linki de mevcut. Sahadaki ekip haritayı açtığında ben neredeyim, etrafımdaki çağrılar neymiş düşüncesiyle zoom in/out yaparak yüzlerce çağrı içerisinde kaybolmasın, kuş uçuşu mesafede kendi konumuna en yakın çağrıyı tek tıkla görerek -eğer yapabiliyorsa- enkaza yönelsin gibi bir fikirdi bu. Yanlış anlaşılma olduğu düşüncesiyle tekrar özetlemek istedim.

Belki hem en yakın hem de en yoğun konumu bul gibi bir şey daha faydalı olabilir. Ya da ileride yapılması düşünülen [filtreler](https://github.com/acikkaynak/deprem-yardim-frontend/discussions/552#discussioncomment-4914102) içerisine yerleştirilebilir.

**Reply by 0xCakin** [reply]
Like the idea. gathering more info. Will be listed in the backlog. @kaganbuzcu
